### PR TITLE
Expand metric collection for HikariCP integration

### PR DESCRIPTION
**Pull request by zbcon**
### What does this PR do?
Adds collection for prometheus-style metrics within the HikariCP integration.

Currently HikariCP is inconsistent in how it names metrics (see [this issue raised in HikariCP](https://github.com/brettwooldridge/HikariCP/issues/1714)). The current datadog integration is designed for collection from Micrometer-style metrics only, it is not collecting the majority of metrics produced in the prometheus-style.

In this PR, we add mapping to collect a variety of metrics in the prometheus-style, as well as extending the current testing to cover these new metrics.

### Motivation
Our use of HikariCP is configured to produce prometheus-style metrics. We want to collect these metrics via the official Datadog integration, but found the `v1.1.1` integration is only built to collect Micrometer-style metrics. The changes in this PR will enable collection of Prometheus-style metrics in addition to Micrometer-style. 

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes
The core issue with the HikariCP integration stems from the inconsistent naming of metrics that HikariCP produces. This is really an issue with HikariCP, has been raised ([issue](https://github.com/brettwooldridge/HikariCP/issues/1714)), and even has a PR to address this issue ([PR](https://github.com/brettwooldridge/HikariCP/pull/1786)). However, these are 3 years old with no recent activity... it doesn't look like HikariCP is going to address this issue anytime soon.

It seems unlikely the core issue is going to be resolved (and is out of our control), so we submit this PR to patch the issue within the integration

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: hikaricp/tests/fixtures/metrics-prometheus.txt
Comment (steveny91) [inline hikaricp/tests/fixtures/metrics-prometheus.txt:6]: Is there a specific reason the fixture here has a few empty lines?

File: hikaricp/CHANGELOG.md
Comment (steveny91) [inline hikaricp/CHANGELOG.md:16]: Thanks for adding these! More of a personal preference here. But I would just go with the first line as a summary for the change log entry:

```suggestion
* Added collection for prometheus-style named metrics.
```

File: hikaricp/tests/fixtures/metrics-prometheus.txt
Comment (zbcon) [reply hikaricp/tests/fixtures/metrics-prometheus.txt:6]: No reason. I must have missed this when tidying up my code 🙈.
I'll remove

Comment (zbcon) [reply hikaricp/tests/fixtures/metrics-prometheus.txt]: This change has been made. Thanks for spotting 👍

File: hikaricp/CHANGELOG.md
Comment (zbcon) [reply hikaricp/CHANGELOG.md]: This change has been made 👍

**Review by Kyle-Neale — COMMENTED** [review]
Hey @zbcon,

Can you just update/rebase your branch so we can get the currently failing test to pass? After that, I believe it should be good to approve. 🙇

**Review by FlorentClarret — APPROVED** [review]
Thanks for the PR. I'll merge once the CI is done!

**Comment by zbcon** [comment]
Thanks very much guys 🖖

**Pull request merged**
